pkg/sql/copy_file_upload.go

```diff
@@ -112,6 +118,7 @@ func (f *fileUploadMachine) run(ctx context.Context) error {
 	err := f.c.run(ctx)
 	_ = f.writeToFile.Close()
 	if err != nil {
+		f.failureCleanup()
```

Comment:
What about if f.writeError has an error? Can/should the API be changed so there's only 1 possible error produced?

Reply:
I will think about restructuring this. However, since this is on two sides of an `io.Pipe`, we still need to capture either side's failures.